docs/tutorials/pytorch.ipynb

```diff
@@ -0,0 +1,507 @@
+    "\n",
+    "train_dataloader = DataLoader(train_dataset, batch_size, shuffle=True)\n",
+    "val_dataloader = DataLoader(val_dataset, batch_size, shuffle=False)\n",
+    "test_dataloader = DataLoader(test_dataset, batch_size, shuffle=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7162d06-8814-4680-8192-aff279e70049",
+   "metadata": {},
+   "source": [
+    "## Transforms\n",
+    "\n",
+    "There are two categories of transforms a user may want to apply to their data:\n",
+    "\n",
+    "* Preprocessing: required to make data \"ML-ready\"\n",
+    "* Data augmentation: designed to artificially inflate the size of the dataset\n",
+    "\n",
+    "Preprocessing transforms such as normalization and one-hot encodings are applied to both training and evaluation data. Data augmentation transforms such as random flip and rotation are typically only performed during training. Below, we initialize transforms for both using the [Kornia](https://kornia.readthedocs.io/en/latest/augmentation.html) library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "efc0152c-e7e4-4f06-9418-9d2c5dd803c3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocess = K.Normalize(0, 10000)\n",
+    "augment = K.ImageSequential(K.RandomHorizontalFlip(), K.RandomVerticalFlip())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e80cda68-19ea-4cd5-a3bc-cf8fcf8147af",
+   "metadata": {},
+   "source": [
+    "## Model\n",
+    "\n",
+    "Our goal is to learn some function $f$ that can map between input $x$ and expected output $y$. Mathematically, this can be expressed as:\n",
+    "\n",
+    "$$x \\overset{f}{\\mapsto} y, \\quad y = f(x)$$\n",
+    "\n",
+    "Since our $x$ in this case is an image, we choose to use ResNet-18, a popular *convolutional neural network* (CNN). We also initialize our model with weights that have been pre-trained on Sentinel-2 imagery so we don't have to start from scratch. This process is known as *transfer learning*."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c7cda7a8-2cd6-46a0-a1c2-bafc751a23f2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = resnet18(ResNet18_Weights.SENTINEL2_ALL_MOCO)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a12b9e9b-26cc-43f4-a517-31b805862df5",
+   "metadata": {},
+   "source": [
+    "## Loss Function\n",
+    "\n",
+    "If $y$ is our expected output (also called \"ground truth\") and $\\hat{y}$ is our predicted output, our goal is to minimize the difference between $y$ and $\\hat{y}$. This difference is referred to as *error* or *loss*, and the loss function tells us how big of a mistake we made. For regression tasks, a simple mean squared error is sufficient:\n",
+    "\n",
+    "$$\\mathcal{L}(y, \\hat{y}) = \\left(y - \\hat{y}\\right)^2$$\n",
+    "\n",
+    "For classification tasks, such as EuroSAT, we instead use a negative log-likelihood:\n",
```

Comment:
MSE is also log-likelihood of a Gaussian with unit variance.

Reply:
Interesting, didn't know that. I think it's easier to explain MSE as is, although it would be nice to tie into NLL/CE more smoothly.